### PR TITLE
#208: Create template

### PR DESCRIPTION
**Pull request by m-markau**
Implementation for task #208

### CODE REVIEW

File: backend/eln/eln-core/src/main/java/com/epam/indigoeln/eln/util/TemplateValidationUtil.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+public class TemplateValidationUtil {
+
+    /**
+     * Validates a template request based on the provided specification.
+     *
+     * @param templateRequest    the template request to validate.
+     * @param templateRepository the template repository.
+     */
+    public static void validateTemplateRequest(TemplateRequest templateRequest, TemplateRepository templateRepository) {
+        if (templateRequest.getName().isBlank()) {
```
Comment (denis-dzenskevich) [inline backend/eln/eln-core/src/main/java/com/epam/indigoeln/eln/util/TemplateValidationUtil.java]: Shall be handled automatically if you put @NotNull on a field and add @Valid on a Controller (Resource) parameter. Same for other empty/blank checks

See e. g. ExperimentServiceTest.testCreateExperimentValidation

File: backend/eln/eln-core/src/main/java/com/epam/indigoeln/eln/service/TemplateService.java [hunk 2]
```diff
@@ -47,10 +54,22 @@ public TemplateDetailsDTO getTemplate(UUID templateId) {
...
     public TemplateDetailsDTO editTemplate(UUID templateId, TemplateEditRequest request) {
         aclService.ensureTopLevelAccess(ApplicationPermission.MANAGE_TEMPLATES);
+
         TemplateEntity template = templateRepository.get(templateId);
-        editProperty(request.getName(), template::setName);
+
+        // Edit the template name if provided
+        editProperty(request.getName(), name -> {
+            // Validate the new name for uniqueness
+            if (templateRepository.existsByName(name)) {
```
Comment (denis-dzenskevich) [inline backend/eln/eln-core/src/main/java/com/epam/indigoeln/eln/service/TemplateService.java]: Optional, but I would rely on a database constraint for that (like in ProjectService.createProject for example)

First, it will work uniformly for new and updated entities. Second, it saves from corner cases. E. g. update with the same name will trigger an error here

File: backend/eln/eln-core/src/test/resources/com.epam.indigoeln.eln.entity/template_entity.json [hunk 3]
```diff
@@ -0,0 +1,67 @@
+{
```
Comment (denis-dzenskevich) [inline backend/eln/eln-core/src/test/resources/com.epam.indigoeln.eln.entity/template_entity.json]: Do we need this file in Git? If it's just to see the output, please generate file somewhere inside "build" directory so it's not included

**Review by denis-dzenskevich — CHANGES_REQUESTED** [review]

File: backend/eln/eln-core/src/main/java/com/epam/indigoeln/eln/util/TemplateValidationUtil.java (hunk 1 above)
Comment (m-markau) [reply backend/eln/eln-core/src/main/java/com/epam/indigoeln/eln/util/TemplateValidationUtil.java]: Done

File: backend/eln/eln-core/src/main/java/com/epam/indigoeln/eln/service/TemplateService.java (hunk 2 above)
Comment (m-markau) [reply backend/eln/eln-core/src/main/java/com/epam/indigoeln/eln/service/TemplateService.java]: Done. Rollback to original version

File: backend/eln/eln-core/src/test/resources/com.epam.indigoeln.eln.entity/template_entity.json (hunk 3 above)
Comment (m-markau) [reply backend/eln/eln-core/src/test/resources/com.epam.indigoeln.eln.entity/template_entity.json]: I put in build directory

File: backend/eln/eln-core/src/main/java/com/epam/indigoeln/eln/service/TemplateService.java (hunk 2 above)
Comment (denis-dzenskevich) [reply backend/eln/eln-core/src/main/java/com/epam/indigoeln/eln/service/TemplateService.java]: Seems it's still checked by existsByName?

Comment (m-markau) [reply backend/eln/eln-core/src/main/java/com/epam/indigoeln/eln/service/TemplateService.java]: I verified it ones more time. 

In latest version of this branch: backend-v3-208-create-template - method rollbacked.

Here is what I see in git:
    public TemplateDetailsDTO editTemplate(UUID templateId, TemplateEditRequest request) {
        aclService.ensureTopLevelAccess(ApplicationPermission.MANAGE_TEMPLATES);

        TemplateEntity template = templateRepository.get(templateId);

        editProperty(request.getName(), template::setName);

        ModelUtil.updateDates(template, userService.getCurrentUser());

        templateRepository.flushAndClear();

        return getTemplate(templateId);
    }

No check for existsByName

Comment (m-markau) [reply backend/eln/eln-core/src/main/java/com/epam/indigoeln/eln/service/TemplateService.java]: I have replaced TemplateRepository.existsByName() by DB unique constraint on "name" field

**Review by denis-dzenskevich — APPROVED** [review]
